mlrun/runtimes/nuclio/api_gateway.py

```diff
@@ -18,33 +18,36 @@
 
 import requests
 from nuclio.auth import AuthInfo as NuclioAuthInfo
-from requests.auth import HTTPBasicAuth
+from nuclio.auth import AuthKinds as NuclioAuthKinds
 
 import mlrun
 import mlrun.common.schemas
+import mlrun.common.types
+from mlrun.model import ModelObj
 from mlrun.platforms.iguazio import min_iguazio_versions
+from mlrun.utils import logger
 
-from ...model import ModelObj
-from ..utils import logger
-from .function import RemoteRuntime, get_fullname, min_nuclio_versions
-from .serving import ServingRuntime
+from .function import get_fullname, min_nuclio_versions
 
-NUCLIO_API_GATEWAY_AUTHENTICATION_MODE_BASIC_AUTH = "basicAuth"
-NUCLIO_API_GATEWAY_AUTHENTICATION_MODE_NONE = "none"
-NUCLIO_API_GATEWAY_AUTHENTICATION_MODE_ACCESS_KEY = "accessKey"
 PROJECT_NAME_LABEL = "nuclio.io/project-name"
 
 
+class APIGatewayAuthenticationMode(mlrun.common.types.StrEnum):
+    NONE = "none"
+    BASIC = "basicAuth"
+    ACCESS_KEY = "accessKey"
```

Comment:
Why not using https://github.com/mlrun/mlrun/blob/87244882c16090977448d7b1500842d3732921fe/mlrun/common/schemas/api_gateway.py#L23?

Reply:
No actual reason :|
removing it